state/protocol/badger/snapshot.go

```diff
@@ -41,6 +41,88 @@ func (s *Snapshot) Head() (*flow.Header, error) {
 	return head, err
 }
 
+// QuorumCertificate returns a valid quorum certificate for the block at the
+// head of this snapshot, if one exists.
+//
+// For root block snapshots, returns the root quorum certificate. For all other
+// blocks, generates a quorum certificate from a valid child, if one exists.
+func (s *Snapshot) QuorumCertificate() (*flow.QuorumCertificate, error) {
+
+	// CASE 1: for the root block, return the root QC
+	root, err := s.state.Params().Root()
+	if err != nil {
+		return nil, fmt.Errorf("could not get root: %w", err)
+	}
+
+	head, err := s.Head()
+	if err != nil {
+		return nil, fmt.Errorf("could not get head: %w", err)
+	}
+	if head.Height == root.Height {
+		// TODO store root QC and return here
+		return nil, fmt.Errorf("root qc not stored")
+	}
+
+	// CASE 2: for any other block, generate the root QC from a valid child
+	child, err := s.validChild()
+	if err != nil {
+		return nil, fmt.Errorf("could not get child: %w", err)
+	}
+
```

Comment:
Better to add a sanity check that the `header.ID() == child.ParentID`

Reply:
👍 Added in https://github.com/onflow/flow-go/pull/201/commits/7f383638d3cf618bf861e01a3e649c6f3b7b64a3